ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -51,6 +51,8 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
     const [nodePanelView, setNodePanelView] = useState(queryParams.get('nodePanelView'));
     const [sidePanel, setSidePanel] = useState(queryParams.get('sidePanel'));
 
+    const isRunningRedirectHandler = useRef(false);
```

Comment:
maybe I'm mistaken, but I don't see this Ref has any functional effect in this code

Reply:
I use it to handle a potential race condition. I think the race condition shouldn't happen 99% of the time and even then, it probably isn't a big issue at all. See https://github.com/argoproj/argo-workflows/pull/7854/files#diff-0670537605e4212d6dada675c6498eabe38d45eb0d3594d59c0779905a83f18bR205

Reply:
Sorry I was in a bit of a rush, I only just checked this properly and I was forgetting to actually update the ref to true.